doc/languages-frameworks/maven.section.md

```diff
@@ -80,360 +80,3 @@ To make sure that your package does not add extra manual effort when upgrading M
-        <version>5.1.1</version>
-      </dependency>
-  </dependencies>
-</project>
-```
-
-Our main class file will be very simple:
-
-```java
-import com.vdurmont.emoji.EmojiParser;
-
-public class Main {
-  public static void main(String[] args) {
-    String str = "NixOS :grinning: is super cool :smiley:!";
-    String result = EmojiParser.parseToUnicode(str);
-    System.out.println(result);
-  }
-}
-```
-
-You find this demo project at [https://github.com/fzakaria/nixos-maven-example](https://github.com/fzakaria/nixos-maven-example).
-
-### Solving for dependencies {#solving-for-dependencies}
-
-#### buildMaven with NixOS/mvn2nix-maven-plugin {#buildmaven-with-nixosmvn2nix-maven-plugin}
-`buildMaven` is an alternative method that tries to follow similar patterns of other programming languages by generating a lock file. It relies on the maven plugin [mvn2nix-maven-plugin](https://github.com/NixOS/mvn2nix-maven-plugin).
-
-First you generate a `project-info.json` file using the maven plugin.
-
-> This should be executed in the project's source repository or be told which `pom.xml` to execute with.
-
-```bash
-# run this step within the project's source repository
-❯ mvn org.nixos.mvn2nix:mvn2nix-maven-plugin:mvn2nix
-
-❯ cat project-info.json | jq | head
-{
-  "project": {
-    "artifactId": "maven-demo",
-    "groupId": "org.nixos",
-    "version": "1.0",
-    "classifier": "",
-    "extension": "jar",
-    "dependencies": [
-      {
-        "artifactId": "maven-resources-plugin",
-```
-
-This file is then given to the `buildMaven` function, and it returns 2 attributes.
-
-**`repo`**:
-    A Maven repository that is a symlink farm of all the dependencies found in the `project-info.json`
-
-
-**`build`**:
-    A simple derivation that runs through `mvn compile` & `mvn package` to build the JAR. You may use this as inspiration for more complicated derivations.
-
-Here is an [example](https://github.com/fzakaria/nixos-maven-example/blob/main/build-maven-repository.nix) of building the Maven repository
-
-```nix
-{ pkgs ? import <nixpkgs> { } }:
-with pkgs;
-(buildMaven ./project-info.json).repo
-```
-
-The benefit over the _double invocation_ as we will see below, is that the _/nix/store_ entry is a _linkFarm_ of every package, so that changes to your dependency set doesn't involve downloading everything from scratch.
-
-```bash
-❯ tree $(nix-build --no-out-link build-maven-repository.nix) | head
-/nix/store/g87va52nkc8jzbmi1aqdcf2f109r4dvn-maven-repository
-├── antlr
-│   └── antlr
-│       └── 2.7.2
-│           ├── antlr-2.7.2.jar -> /nix/store/d027c8f2cnmj5yrynpbq2s6wmc9cb559-antlr-2.7.2.jar
-│           └── antlr-2.7.2.pom -> /nix/store/mv42fc5gizl8h5g5vpywz1nfiynmzgp2-antlr-2.7.2.pom
-├── avalon-framework
-│   └── avalon-framework
-│       └── 4.1.3
-│           ├── avalon-framework-4.1.3.jar -> /nix/store/iv5fp3955w3nq28ff9xfz86wvxbiw6n9-avalon-framework-4.1.3.jar
-```
-
-#### Double Invocation {#double-invocation}
```

Comment:
I think I wrote this explanation section; I think it's all pretty good advice still on learning lessons on how to integrate java with Nix.

Maybe we can retitle these sections and just remove the top.